cmd/algorand-indexer/daemon.go

```diff
@@ -315,61 +324,86 @@ func runDaemon(daemonConfig *daemonConfig) error {
 		daemonConfig.noAlgod = true
 	}
 	opts := idb.IndexerDbOptions{}
-	if daemonConfig.noAlgod && !daemonConfig.allowMigration {
-		opts.ReadOnly = true
-	}
+	// if daemonConfig.noAlgod && !daemonConfig.allowMigration {
+	opts.ReadOnly = true
+	// }
 
 	opts.MaxConn = daemonConfig.maxConn
 	opts.IndexerDatadir = daemonConfig.indexerDataDir
 	opts.AlgodDataDir = daemonConfig.algodDataDir
 	opts.AlgodToken = daemonConfig.algodToken
 	opts.AlgodAddr = daemonConfig.algodAddr
 
-	db, availableCh := indexerDbFromFlags(opts)
-	defer db.Close()
 	var wg sync.WaitGroup
 	if bot != nil {
 		wg.Add(1)
-		go runBlockImporter(ctx, daemonConfig, &wg, db, availableCh, bot, opts)
+		logger.Info("Initializing exporter.")
+		// Need to update config file w/ options provided via CLI
+		writeCLIToPlugin(daemonConfig)
+		exp, err := exporters.ExporterByName(daemonConfig.exporterPlugin, daemonConfig.indexerDataDir, logger)
+		if err != nil {
+			maybeFail(err, "unable to construct requested Exporter: %v", err)
+		}
+		go runBlockImporter(ctx, daemonConfig, &wg, bot, opts, exp)
 	} else {
 		logger.Info("No block importer configured.")
 	}
 
-	fmt.Printf("serving on %s\n", daemonConfig.daemonServerAddr)
-	logger.Infof("serving on %s", daemonConfig.daemonServerAddr)
-
-	options := makeOptions(daemonConfig)
-
-	api.Serve(ctx, daemonConfig.daemonServerAddr, db, bot, logger, options)
+	// Only serve the API for the postgres exporter
+	if daemonConfig.exporterPlugin == postgresql.ExporterName {
+		fmt.Printf("serving on %s\n", daemonConfig.daemonServerAddr)
+		logger.Infof("serving on %s", daemonConfig.daemonServerAddr)
+		db, availableCh := indexerDbFromFlags(opts)
+		defer db.Close()
+		options := makeOptions(daemonConfig)
+		// Wait for db to become available
+		<-availableCh
+		api.Serve(ctx, daemonConfig.daemonServerAddr, db, bot, logger, options)
+	}
 	wg.Wait()
 	return err
 }
 
-func runBlockImporter(ctx context.Context, cfg *daemonConfig, wg *sync.WaitGroup, db idb.IndexerDb, dbAvailable chan struct{}, bot fetcher.Fetcher, opts idb.IndexerDbOptions) {
+// Bridges functionality between setting config via CLI and the new plugin configs
+func writeCLIToPlugin(cfg *daemonConfig) {
+	switch cfg.exporterPlugin {
+	case postgresql.Metadata.Name():
+		pluginCfg := plugins.LoadConfig(logger, cfg.indexerDataDir, postgresql.Metadata)
+		var expCfg postgresql.ExporterConfig
+		err := yaml.Unmarshal([]byte(pluginCfg), &expCfg)
```

Comment:
nit: maybe use this pattern
```suggestion
		if err := yaml.Unmarshal([]byte(pluginCfg), &expCfg); err != nil {
			maybeFail(err, "unable to load postgresql config: %v", err)
		}
```